crates/bevy_ecs/src/storage/table.rs

```diff
@@ -240,27 +288,55 @@ impl Column {
         self.changed_ticks.reserve_exact(additional);
     }
 
+    /// Fetches the data pointer to the first element of the [`Column`].
+    ///
+    /// The pointer is type erased, so using this function to fetch
     #[inline]
     pub fn get_data_ptr(&self) -> Ptr<'_> {
         self.data.get_ptr()
     }
 
+    /// Fetches the slice to the [`Column`]'s data cast to a given type.
+    ///
+    /// Note: The values stored within are [`UnsafeCell`].
+    /// Users of this API must ensure that accesses to each individual element
+    /// adhere to the safety invariants of [`UnsafeCell`].
```

Comment:
I don't know who created the practice of writing down safety invariants but I really like the practice. It's fascinating to see coming from a higher level language that this is done in Rust apparently.

Reply:
Yeah this is ecosystem wide. It's really lovely.